lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -102,12 +102,24 @@ class VaultClientError(Exception):
 class Vault:
     """Class to interact with Vault through its API."""
 
-    def __init__(self, url: str, ca_cert_path: str):
-        self._client = hvac.Client(url=url, verify=ca_cert_path)
+    def __init__(self, url: str, ca_cert_path: Optional[str]):
+        self._client = hvac.Client(url=url, verify=ca_cert_path if ca_cert_path else False)
 
-    def authenticate(self, auth_details: AuthMethod) -> None:
+    def authenticate(self, auth_details: AuthMethod) -> bool:
         """Find and use the token related with the given auth method."""
-        auth_details.login(self._client)
+        try:
+            auth_details.login(self._client)
+        except (Forbidden, InternalServerError):
```

Comment:
Done

Reply:
Wouldn't it be useful to log the actual exception too? This would be valid for the whole lib.

Reply:
I added that here, but I don't think it would be useful to add it to the ones where we already check what the message says. Forbidden just returns a really generic permission denied string, and the already enabled exceptions also say very generic things like request failed.